File: apps/ui-kit/src/lib/utils/format.ts
// Copyright (c) 2024 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

export function truncate(value: string, startLength: number = 6, endLength: number = 6): string {
    return value.length > startLength + endLength
        ? `${value.slice(0, startLength)}...${value.slice(-endLength)}`
        : value;
}

File: crates/iota-core/tests/format.rs
// Copyright (c) 2021, Facebook, Inc. and its affiliates
// Copyright (c) Mysten Labs, Inc.
// Modifications Copyright (c) 2025 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

#[test]
#[cfg_attr(msim, ignore)]
fn test_format() {
    // If this test breaks and you intended a format change, you need to run to get the fresh format:
    // # cargo -q run --example generate-format -- print > crates/iota-core/tests/staged/iota.yaml

    let status = std::process::Command::new("cargo")
        .current_dir("..")
        .args(["run", "--example", "generate-format", "--"])
        .arg("test")
        .status()
        .expect("failed to execute process");
    assert!(
        status.success(),
        "\n\
If this test breaks and you intended a format change, you need to run to get the fresh format:\n\
cargo -q run --example generate-format -- print > crates/iota-core/tests/staged/iota.yaml\n\
        "
    );
}
